galasa-simbank-tests/dev.galasa.simbank.tests/src/main/java/dev/galasa/simbank/tests/WebAppIntegrationTest.java

```diff
@@ -0,0 +1,148 @@
+    
+    @ZosBatchJobname(imageTag="SIMBANK")
+	public IZosBatchJobname zosBatchJobname;
+	
+	@CoreManager
+    public ICoreManager     coreManager;
+	
+	@BundleResources
+	public IBundleResources resources; 
+
+	@SeleniumManager
+    public ISeleniumManager seleniumManager;
+
+	@Logger
+	public Log logger;
+	
+	public static int account;
+	
+	@Test
+	public void webAppIntegrationTest() throws SeleniumManagerException, DockerManagerException, TimeoutException, KeyboardLockedException, TerminalInterruptedException, TextNotFoundException, FieldNotFoundException, NetworkException, TestBundleResourceException, IOException, ZosBatchException{
+
+		//Dianas code
+		terminal.waitForKeyboard()
+		.positionCursorToFieldContaining("Userid").tab().type("IBMUSER")
+		.positionCursorToFieldContaining("Password").tab().type("SYS1")
+		.enter().waitForKeyboard()
+
+        // Open banking application
+		.pf1().waitForKeyboard()
+		.clear().waitForKeyboard().type("bank")
+		.enter().waitForKeyboard();
+
+		String accountNumber = String.valueOf((int)Math.random() * (999999999 - 100000000 + 1) + 100000000);
+
+		terminal.pf1().waitForKeyboard()
+		.positionCursorToFieldContaining("Account Number").tab().type(accountNumber)
+		.enter().waitForKeyboard();
+		
+		
+		while(!terminal.retrieveScreen().contains("Account Not Found")){
+			
+		accountNumber = String.valueOf((int)Math.random() * (999999999 - 100000000 + 1) + 100000000);
+
+		terminal.pf3().pf1().waitForKeyboard()
+		.positionCursorToFieldContaining("Account Number").tab().type(accountNumber)
+		.enter().waitForKeyboard();
+		}
+		
+		HashMap<String, Object> parameters = new HashMap<>();
+        parameters.put("CONTROL", "ACCOUNT_OPEN");
+		parameters.put("DATAIN", accountNumber+",00-00-00,1");	
+		String jcl = resources.retrieveSkeletonFileAsString("/resources/skeletons/SIMBANK.jcl", parameters);
+		IZosBatchJob batchJob = zosBatch.submitJob(jcl, zosBatchJobname);
+		logger.info("batchJob.toString() = " +  batchJob.toString());
+		int rc = batchJob.waitForJob();
+		if (rc != 0) {
+			// Print the job output to the run log
+			batchJob.retrieveOutput().forEach(jobOutput ->
+				logger.info("batchJob.retrieveOutput(): " + jobOutput.getDdname() + "\n" + jobOutput.getRecords() + "\n")
+			);
+			Fail.fail("Batch job failed RETCODE=" + batchJob.getRetcode() + " Check batch job output");
+			
+		}
+		logger.info("Batch job complete RETCODE=" + batchJob.getRetcode());
+		
+		//Dianas code
+		
+		//Charlies code
+		
+		Map<String, List<InetSocketAddress>> ports = container.getExposedPorts();
+		
+		int port = ports.get("8080/tcp").get(0).getPort();
+
+		String webpage = "http://localhost:" + port + "galasa-simplatform-webapp";
+
+		IWebPage page = seleniumManager.allocateWebPage(webpage);
+		page.maximize();
+		assertThat(page.getTitle()).containsOnlyOnce("Simbank");
+		page.sendKeysToElementById("accnr", accountNumber);
+		page.sendKeysToElementById("amount", "120");
+		page.clickElementById("submit");
```

Comment:
Can we take some screenshots of the page we are navigating so that they get stored in the RAS.